pkg/apis/kubeone/types.go

```diff
@@ -395,8 +399,27 @@ type ClusterNetworkConfig struct {
 
 	// KubeProxy config
 	KubeProxy *KubeProxyConfig `json:"kubeProxy,omitempty"`
+
+	// IPFamily "" | IPv4 | IPv6 | IPv4+IPv6.
+	IPFamily IPFamily `json:"ipFamily,omitempty"`
+
+	// NodeCIDRMaskSizeIPv4
+	NodeCIDRMaskSizeIPv4 *int `json:"nodeCIDRMaskSizeIPv4,omitempty"`
+
+	// NodeCIDRMaskSizeIPv6
+	NodeCIDRMaskSizeIPv6 *int `json:"nodeCIDRMaskSizeIPv6,omitempty"`
 }
 
+// IPFamily IPv4 | IPv6 | IPv4+IPv6.
+type IPFamily string
+
+const (
+	Unspecified IPFamily = "" // interpreted as IPv4
+	IPv4        IPFamily = "IPv4"
+	IPv6        IPFamily = "IPv6"
```

Comment:
Do we actually support this case or only dual-stack?

Reply:
Only dualstack. That too `IPv4+IPv6` and not `IPv6+IPv4` kind. 

Can get rid of it if you prefer to add it when we actually support it.

We do it like this in KKP and MC.

Reply:
Let's get rid of types that we don't use and/or support (for now).